storage-client/src/client.rs

```diff
@@ -32,15 +45,123 @@ impl StorageClientImpl {
         })
     }
+
+    /// Returns the physical location of the requested data in RequestParams.
+    async fn get_info_from_catalog(&self, _request: StorageRequest) -> Result<RequestParams> {
+        // FIXME (kunle): Need to discuss with the catalog team.
+        // The following line serves as a placeholder for now.
+        Ok(RequestParams::File("userdata1.parquet".to_string()))
+    }
 }
 
 #[async_trait::async_trait]
 impl StorageClient for StorageClientImpl {
     async fn request_data(&self, _request: StorageRequest) -> Result<Receiver<RecordBatch>> {
-        todo!()
+        // First we need to get the location of the parquet file from the catalog server.
+        let _location = match self.get_info_from_catalog(_request).await? {
+            RequestParams::File(location) => location,
+            _ => {
+                return Err(anyhow!(
+                    "Failed to get location of the file from the catalog server."
+                ));
+            }
+        };
+
+        // Then we need to send the request to the storage server.
+        let url = format!("{}file/{}", self._storage_server_endpoint, _location);
+        let client = Client::new();
+        println!("Requesting data from: {}", url);
+        let response = client.get(url).send().await?;
+        if response.status().is_success() {
+            // Store the streamed Parquet file in a temporary file.
+            // FIXME: 1. Do we really need streaming here?
+            //       2. Do we need to store the file in a temporary file?
+            let temp_dir = tempdir()?;
+            let file_path = temp_dir.path().join("tmp.parquet");
+            let mut file = File::create(&file_path)?;
+            let mut stream = response.bytes_stream();
+            while let Some(chunk) = stream.next().await {
+                let chunk = chunk?;
+                file.write_all(&chunk)?;
+            }
+
+            // Convert the Parquet file to a record batch.
+            let file = AsyncFile::open(file_path).await.unwrap();
+            let builder = ParquetRecordBatchStreamBuilder::new(file)
+                .await
+                .unwrap()
+                .with_batch_size(BATCH_SIZE);
+
+            let mask = ProjectionMask::all();
+            let stream = builder.with_projection(mask).build().unwrap();
+            let results = stream.try_collect::<Vec<_>>().await.unwrap();
+
+            // Return the record batch as a stream.
+            let (tx, rx) = channel(CHANNEL_CAPACITY);
+            tokio::spawn(async move {
+                for result in results {
+                    tx.send(result).await.unwrap();
+                }
+            });
+            Ok(rx)
+        } else {
+            Err(anyhow::anyhow!(
+                "Failed to download file. Response: {:?}",
+                response.status()
+            ))
+        }
     }
 
+    // TODO (kunle): I don't think this function is necessary.
+    #[cfg(feature = "ignore")]
```

Comment:
I don't think rust can ignore a method by adding this. You can refer to https://doc.rust-lang.org/reference/conditional-compilation.html for conditional compilation but I don't think we need to do conditional compilation here?